lib/flipper/adapters/active_record.rb

```diff
@@ -244,17 +245,26 @@ def delete(feature, gate)
       end
 
       def enable_multi(feature, gate, thing)
-        with_connection(@gate_class) do
-          @gate_class.create! do |g|
-            g.feature_key = feature.key
-            g.key = gate.key
-            g.value = thing.value.to_s
+        with_connection(@gate_class) do |connection|
+          begin
+            connection.transaction do
+              savepoint_name = "flipper_savepoint_#{SecureRandom.hex(8)}"
+              connection.execute("SAVEPOINT #{savepoint_name}")
```

Comment:
You can use `transaction(require_new: true) do` https://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/DatabaseStatements.html#method-i-transaction

Reply:
And that's also probably why you are "poisoning" the connection. `conn.transaction { conn.transaction { } }` doesn't do nested transaction. You need `require_new: true` for that.

Reply:
Awesome! I'll try that out. Thanks for taking the time to teach!

Reply:
https://github.com/flippercloud/flipper/pull/881/commits/1c9e0a295d20de1b4fd58d94616651dbbdcb2b83 did the trick. 

Still odd to me that mysql and sqlite don't raise about the poisoned transaction and pg gem does. But perhaps it is something in the pg gem and not the AR adapter for Postgres? 

`requires_new: true` is definitely cleaner than the save point junk I had so I'll go with that for now.